includes/panopto_derivatives.inc

```diff
@@ -5,37 +5,386 @@
+          'message_sub' => array(
+            '@endpoint' => $endpoint,
+            '@response' => $response,
+          ),
+          'type' => 'watchdog',
+          'severity' => WATCHDOG_ERROR,
+        );
+        $failureMessage['messages'][] = $messages;
+        watchdog($messages);
+      }
+    } else {
+      $messages = array(
+        'message' => t("Expected one of @expected; got a @response_code"),
+        'message_sub' => array(
+          '@expected' => implode(',', array_keys($expect)),
+          '@response_code' => $response_code,
+        ), 
+        'type' => 'watchdog',
+        'severity' => WATCHDOG_ERROR,
+      );
+      $failureMessage['messages'][] = $messages;
+      watchdog($messages);
+    }
+  }
+  return;
+}
+
+/**
+ * Uploads file to Panopto using AWS S3 Uploader
+ * @param $upload_target string The URL of upload target
+ * @param $file_path string The URL which is the target of the request
+ * @return void
+ */
+function islandora_video_streaming_panopto_upload_file($upload_target, $file_path) {
+  $element = explode("/", $upload_target);
+  $prefix = array_pop($element);
+  $service_endpoint = implode("/", $element);
+  $bucket = array_pop($element);
+  $file_name = basename($file_path);
+  $object_key = $prefix."/".$file_name;
+  
+  $s3Params = array(
+      'endpoint' => $service_endpoint,
+      'region'  => 'us-east-1',
+      'version' => '2006-03-01',
+      'credentials' => [
+        'key'    => 'dummy',
+        'secret' => 'dummy'
+      ]
+  );
+  // Create an S3Client
+  $s3Client = new S3Client($s3Params);
+
+  $source = fopen($file_path, 'rb');
+  $uploader = new ObjectUploader(
+    $s3Client,
+    $bucket,
+    $object_key,
+    $source
+  );
+
+  do {
+    try {
+      $result = $uploader->upload();
+    } catch (MultipartUploadException $e) {
+      rewind($source);
+      $uploader = new MultipartUploader($s3Client, $source, [
+        'state' => $e->getState(),
+      ]);
+    }
+  } while (!isset($result));
+}
+
+/**
+ * Creates manifest file, in temporary storage, needed for Panopto Video Upload
+ * @param $file_path The filepath of the Islandora object
+ * @return string The filename or filepath of the newly created manifest file
+ */
+function islandora_video_streaming_panopto_create_manifest_for_video($file_path = null) {
+  $file_name = basename($file_path);
+  $manifest_template = module_load_include('xml', 'islandora_video_streaming', 'includes/upload_manifest_template');
+  $template = file_get_contents($manifest_template);
+  $template = str_replace("{Title}", $file_name, $template);
+  $template = str_replace("{Description}", 'This is a video session with the uploaded video file '.$file_name, $template);
+  $template = str_replace("{Filename}", $file_name, $template);
+  $template = str_replace("{Date}", date('c'), $template);
+  
+  $part_of_file_name = explode(".", $file_name);
+  $updated_manifest_file = "/tmp/$part_of_file_name[0].xml";
```

Comment:
Use Drupal temporary file handling to get a temporary file path that will work across operating systems, and across configured Drupal environments.  You are not guaranteed to have access or write access, to /tmp.

For example: https://github.com/ulsdevteam/islandora_video_streaming/blob/4c8ca2f0f0e9b8eb4d117cdd6b55f67e804a767b/includes/panopto_derivatives.inc#L107